### PR TITLE
CAMEL-15617: Fix filter Google PubSub reserved attributes

### PR DESCRIPTION
**Pull request by alvinkwekel**
When Camel receives a PubSub message with reserved Google PubSub attributes these are set on a Camel header and passed on to any message sender. These reserved attributes are not allowed to be set on an outgoing message so this throws an error.

This filters out these reserved attributes before sending.

https://issues.apache.org/jira/browse/CAMEL-15617

### CODE REVIEW

File: components/camel-google-pubsub/pom.xml
Comment (oscerd) [inline components/camel-google-pubsub/pom.xml:50]: Are you sure about removing the exclusions?

File: parent/pom.xml
Comment (oscerd) [inline parent/pom.xml:208]: I think this require a check on the karaf feature if needed

File: parent/pom.xml
Comment (oscerd) [inline parent/pom.xml:218]: I think that's the reason for removing the exclusions, but we need to double check

**Review by oscerd — CHANGES_REQUESTED** [review]
Just some doubts. Thanks for the PR.

File: components/camel-google-pubsub/pom.xml
Comment (alvinkwekel) [reply components/camel-google-pubsub/pom.xml:50]: PubSub 1.108.1 has a dependency on GRPC 1.57.1 and it doesn't work with the GRPC 1.28.0 as defined in the parent.
Should I increase the version in the parent, add a pubsub specific version in the parent of leave it like so?

Comment (oscerd) [reply components/camel-google-pubsub/pom.xml:50]: Do you mean 1.27.1? I think it makes sense to add a property placeholder only for pubsub and use it in this component only. We can do the same for the karaf feature. Camel-grpc is based on 1.28.0 so we cannot generally downgrade

File: parent/pom.xml
Comment (alvinkwekel) [reply parent/pom.xml:208]: You mean in the camel-dependencies project right? I've updated it there now.

File: parent/pom.xml
Comment (alvinkwekel) [reply parent/pom.xml:218]: Yes that is correct, see my comment above.

File: parent/pom.xml
Comment (oscerd) [reply parent/pom.xml:208]: We need to check the camel-googke-pubsub karaf feature in the camel-karaf repo, but we can do that later

File: components/camel-google-pubsub/pom.xml
Comment (alvinkwekel) [reply components/camel-google-pubsub/pom.xml:50]: Sorry, GRPC 1.30.2. So that would mean a general upgrade. Would that be possible?

Comment (oscerd) [reply components/camel-google-pubsub/pom.xml:50]: I don't think camel-grpc is ready for this. So I believe it would be better to use a specific version only in this component

Comment (alvinkwekel) [reply components/camel-google-pubsub/pom.xml:50]: I've created an explicit dependency on GRPC 1.30.2 now.

**Review by oscerd — APPROVED** [review]
LGTM, we can merge this and then check the karaf feature. Do you have time to add a test or an integration test? This may help when we'll upgrade again.

**Comment by alvinkwekel** [comment]
Perhaps its better to keep Google PubSub at 1.105.0 for now because that uses the same GRPC version as the Camel GRPC component.

**Review by oscerd — APPROVED** [review]
Thank you for this.

**Comment by oscerd** [comment]
Thanks, I'll merge this today.

**Review by omarsmak — APPROVED** [review]
LGTM! 
By the way @alvinkwekel , was this modification tested against a live pubsub instance? The reason is, we had previously some issues whereby some modifications broke the component although was still working against testcontainer instance.

**Comment by oscerd** [comment]
Let me wait for the answer before merging then.

**Comment by alvinkwekel** [comment]
Yes, all tests pass when run against the Cloud PubSub. Although I did notice exceptions being thrown when running the suite: about GRPC channels not being shutdown properly. So I added an additional fix to shutdown all cached producers when the component is shutdown. This solves these exceptions.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**
